lib/app.rb

```diff
@@ -43,44 +43,63 @@ def create_person
 
   def create_student
     print 'Age: '
-    age = gets.chomp.to_i
+    age = Integer(gets.chomp)
+    raise 'Age should be positive.' if age <= 0
 
     print 'Name: '
-    name = gets.chomp
+    name = gets.chomp.strip
+    raise 'Name cannot be empty.' if name.empty?
 
     print 'Has parent permission? [Y/N]: '
-    parent_permission = gets.chomp.downcase == 'y'
+    parent_permission_input = gets.chomp.downcase
+    raise 'Invalid input. Choose Y or N.' unless %w[y n].include?(parent_permission_input)
+
+    parent_permission = parent_permission_input == 'y'
 
     student = Student.new(age, name, parent_permission: parent_permission)
     @people << student
     puts 'Student created successfully!'
+  rescue StandardError => e
+    puts "Error: #{e.message}"
+    retry
   end
 
   def create_teacher
     print 'Age: '
-    age = gets.chomp.to_i
+    age = Integer(gets.chomp)
+    raise 'Age should be positive.' if age <= 0
 
     print 'Name: '
-    name = gets.chomp
+    name = gets.chomp.strip
+    raise 'Name cannot be empty.' if name.empty?
 
     print 'Specialization: '
-    specialization = gets.chomp
+    specialization = gets.chomp.strip
+    raise 'Specialization cannot be empty.' if specialization.empty?
 
     teacher = Teacher.new(age, specialization, name)
```

Comment:
Great job implementing the Teacher.new method however please be informed that this line of code `should've been placed in the 'create_person()' method`. Here's an example code snippet:

```ruby
class App
  def create_person
    choice = UserInput.get_choice

    case choice
    when '1'
      age, name, parent_permission = create_student
      @people << Student.new(age, name, parent_permission: parent_permission)
    when '2'
      age, name, specialization = create_teacher
      @people << Teacher.new(age, specialization, name)
    else
      puts 'Invalid option.'
    end

    puts 'Person created successfully!'
  end

  def create_student
    age = UserInput.get_age
    raise 'Age should be positive.' if age <= 0

    name = UserInput.get_name
    raise 'Name cannot be empty.' if name.empty?

    parent_permission_input = UserInput.get_parent_permission
    raise 'Invalid input. Choose Y or N.' unless %w[y n].include?(parent_permission_input)

    parent_permission = parent_permission_input == 'y'

    [age, name, parent_permission]
  end

  def create_teacher
    age = UserInput.get_age
    raise 'Age should be positive.' if age <= 0

    name = UserInput.get_name
    raise 'Name cannot be empty.' if name.empty?

    specialization = UserInput.get_specialization
    raise 'Specialization cannot be empty.' if specialization.empty?

    [age, name, specialization]
  end
end
```

Reply:
thank for the tips